### PR TITLE
Testing Your Ruby Code - Enumerables

### PR DESCRIPTION
**Pull request by pipe2442**
Ruby - Testing Your Ruby Code - Enumerables

Initiated rspec testing on root rspec --init.
Created testing_rspec branch for testing purposes.
Created enumerables_spec.rb file for testing purposes.
Created test scenarios to validate our code with each enumerable method.

### CODE REVIEW

File: spec/enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  let(:search) { proc { |memo, word| memo.length > word.length ? memo : word } }
+  let(:my_proc) { proc { |i| i * i } }
+  let(:false_block) { proc { |num| num > HIGHEST_VALUE } }
+  let(:block) { proc { |num| num < (LOWEST_VALUE + HIGHEST_VALUE) / 2 } }
+
+  describe '#my each' do
+    it 'Loop through the array and multiply each element' do
+      expect(array2.my_each { |i| p i }).to be(array2.each { |i| p i })
+    end
+  end
```
Comment (javitocor) [inline spec/enumerables_spec.rb]: - test what happens when there is no block given
- test when self is a range

File: spec/enumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  describe '#my_select' do
+    it 'Select even numbers' do
+      expect(array2.my_select(&:even?)).to eq(array2.select(&:even?))
+    end
+    it 'My_select range block should be equal to select block' do
+      expect(range.my_select(&block)).to eq(range.select(&block))
+    end
+  end
```
Comment (javitocor) [inline spec/enumerables_spec.rb]: - test what happens when there is no block given

File: spec/enumerables_spec.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    it 'Loop through the array and multiply each element' do
+      expect(array2.my_each { |i| p i }).to be(array2.each { |i| p i })
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'Print the actual element and its index' do
+      expect(array2.my_each_with_index { |e, i| p e, i }).to be(array2.each_with_index { |e, i| p e, i })
+    end
+  end
```
Comment (javitocor) [inline spec/enumerables_spec.rb]: - test what happens when there is no block given
- test when self is a range

File: spec/enumerables_spec.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    it 'Checks if there is a false array' do
+      expect(false_array.my_any?).to eq(false_array.any?)
+    end
+  end
+
+  describe '#my_none?' do
+    it 'It checks if the current element is a symbol' do
+      expect(array2.my_none? { |e| e.is_a?(Symbol) }).to eq(array2.none? { |e| e.is_a?(Symbol) })
+    end
+  end
```
Comment (javitocor) [inline spec/enumerables_spec.rb]: - Test all the cases requested by TSE when you submitted the enumerables project, i.e. no block given, class, regex, pattern...

File: spec/enumerables_spec.rb [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+  describe '#my_count' do
+    it 'It counts how many elements correspond to the given argument' do
+      expect(array2.my_count(2)).to eq(array2.count(2))
+    end
+  end
```
Comment (javitocor) [inline spec/enumerables_spec.rb]: - test what happens when there is no block given
- test when self is a range

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUIRED 🛑 

Hello team! 👋 
Nice set up of rspec and a good start for this project 👍 
But we still need to implement some changes to get this project approved 💪 
Kindly check my comments below 👇 apply the changes and resubmit your project for a new code review ⏩ 
Good luck! 😺 
## Mandatory changes
- Remember to test positive and negative scenarios
- The testing must provide a way to be sure the methods you created work exactly like the originals, basically you have to test all the cases requested by TSE in your enumerables project, check my comments in every method

**Comment by gscarv13** [comment]
### Thanks for your review! 
All requested changes were revised and fixed accordingly.

**Review by MahmoudBakr23 — APPROVED** [review]
### **Approved 👍**

<hr>

Hello team, great work you've all done in this project it seems professional and complete to me, good work also in the requested changes.
go ahead and merge the pull request 🎉👌🏻

<hr>

Good luck in the next and happy coding 🖥

**Pull request merged**